### ISSUE TITLE
Empty variable in specific format XXXXe

### ISSUE THREAD

**Issue by olegsa** [hashicorp/terraform#13967]
### Terraform Version
Terraform v0.9.3

### Affected Resource(s)
Any resource using ```variables```

### Terraform Configuration Files
```hcl
variable "name" {}

provider "aws" {
  region = "eu-west-1"
}

resource "aws_instance" "web" {
  ami           = "ami-09447c6f"
  instance_type = "m3.large"

  tags {
    Name = "${var.name}"
    Name1 = "102030e"
  }
}

```

### Debug Output
plan: https://gist.github.com/olegsa/ceb2d8039a619c52dd33435987238ddf
apply: https://gist.github.com/olegsa/b3effce9c9836ffe243c0cfb2588f656
### Expected Behavior
What should have happened?

### Actual Behavior
When passing variable via cli in format `XXXXXe` (X - is a digit, no mater what length ) 
I am getting empty string, if it's static string everything works fine 
**CLI:**
```terraform apply -var name=102030e```
**Config:**
```
    Name = "${var.name}"
    Name1 = "102030e"
```
**tfstate:**
```
    tags.Name:                   ""
    tags.Name1:                  "102030e"
```
Same happens not only in tags , also in names of Autoscaling, Launch Configuration etc ...
Looks like problem with parsing specific format `XXXXe`

**Comment by vtorhonen** [comment]
Try to define the variable as a string like this: `terraform apply -var 'name="102030e"'`.

**Comment by olegsa** [comment]
@vtorhonen , yes its work 
but if I cannot use it in this case 
```sh
myvar="12345e"
terraform apply -var name=$myvar
```

**Comment by vtorhonen** [comment]
@olegsa You could then use something like this:

```
myvar="12345e"
terraform apply -var name="'${myvar}'"
```

**Comment by olegsa** [comment]
@vtorhonen 
```
myvar="12345e"
terraform apply -var name="'${myvar}'"
```
Output: ``` 
tags.Name:                   "'12345e'"
tags.Name1:                  "102030e"
```

**Comment by sergeylanzman** [comment]
same issue https://github.com/hashicorp/terraform/issues/17396

**Comment by apparentlymart** [comment]
Hi all! Sorry for the long silence here.

It turns out that this same issue was being tracked in multiple places. Over in #15626 I verified this fixed on master using the v0.12.0-alpha2 prerelease build, so I'm going to close this out too. Please see my comment on the other issue for details on how I verified it.
